refactor(ast): rename `#[estree(add_entry)]` to `#[estree(add_fields)]`

overlookmotel opened the pull request:
#8921 introduced attribute to add extra fields to a struct in ESTree AST `#[estree(add_entry(...))]`. Rename it to `#[estree(add_fields(...))]`, as that's more descriptive of what it does.

overlookmotel commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8926?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#8926** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8926?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8926?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#8925** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8925?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#8924** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8924?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#8923** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8923?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=8926).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

overlookmotel commented:
@hi-ogawa It'd be ideal to specify the name, value, and TS type for extra fields in one go. But I can't see a good way to encode that in attributes right now. We can revisit this. But regardless, the functionality provided by `#[estree(add_fields)]` is good.

Boshen commented:
### Merge activity

* **Feb 6, 10:07 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
refactor(ast): rename `#[estree(add_entry)]` to `#[estree(add_fields)]` (#8926)

#8921 introduced attribute to add extra fields to a struct in ESTree AST `#[estree(add_entry(...))]`. Rename it to `#[estree(add_fields(...))]`, as that's more descriptive of what it does.
